providers/writing_vcr_specs.md

```diff
@@ -0,0 +1,136 @@
+## Writing VCR Provider Spec Tests
+
+So you've just done a refresh with your new provider, congratulations.  Now it is time to make sure it continues to work by writing some tests.
+
+### VCR
+
+How do you test code which makes API calls without having access to a live system?  There is a ruby gem called VCR which "records" HTTP API calls and then plays them back later.  This allows us to test code which would normally have to point to a live API without having to stub every single method call.
+
+The results of the recording are stored in the VCR `config.cassette_library_dir` as YAML files with request/response pairs.  VCR will automatically stub the HTTP layer for you, and when it recognizes a request it will replay the saved response for you.  This allows for mostly normal operation while running in an isolated CI system.
+
+### Configuration
+
+If you generated your provider plugin with the `--vcr` flag then the VCR configuration should have already been done for you.  If not simply add the following to the bottom of your `spec/spec_helper.rb` file:
+```ruby
+VCR.configure do |config|
+  config.ignore_hosts 'codeclimate.com' if ENV['CI']
+  config.cassette_library_dir = File.join(ManageIQ::Providers::AwesomeCloud::Engine.root, 'spec/vcr_cassettes')
+end
+```
+
+The next thing we have to take care of is hiding "secrets".  Since the VCR YAML files will be committed to source control it is critical that private information like passwords do not make it into these files.
+
+VCR handles this with the `config.define_cassette_placeholder` option.  You provide VCR with a string that you want to be replaced, and then what you want it to be replaced with.  This allows for hostanmes / passwords / etc... to be used when recording the cassette but the values will not be written to the resulting YAML files.
+
+ManageIQ has a pattern to help you with this, simply create a `config/secrets.defaults.yml` file:
+```yaml
+---
+test:
+  awesome_cloud_defaults: &awesome_cloud_defaults
+    access_key: AWESOME_CLOUD_ACCESS_KEY
+    secret_key: AWESOME_CLOUD_SECRET_KEY
+  awesome_cloud:
+    <<: *awesome_cloud_defaults
+```
+
+Then create a `config/secrets.yml` file (this file will not be committed and should be in your .gitignore):
+```yaml
+---
+test:
+  awesome_cloud:
+    access_key: "YOUR_REAL_ACCESS_KEY"
+    secret_key: "YOUR_REAL_SECRET_KEY"
+```
+
+Then add the following to your `VCR.configure` block in `spec/spec_helper.rb`:
```

Comment:
You might want to minimize this to only the "new" parts (thought I understand why you might want it all for context).

Reply:
Yeah I thought about that but didn't want people just putting it at the end of the file outside of the block but guess there is just as good a chance they'll duplicate the whole thing.  I'll add a comment about what line specifically to add it after